README.md

```diff
@@ -10,103 +10,150 @@
-
-7 directories, 11 files
+│   ├── v0.4.9
+│   │   ├── build-info
+│   │   ├── build-log-freebsd-386
+│   │   ├── build-log-freebsd-arm
+│   │   ├── dist.json
+│   │   ├── go-ipfs_v0.4.9_darwin-386.tar.gz
+│   │   ├── go-ipfs_v0.4.9_darwin-amd64.tar.gz
+│   │   ├── go-ipfs_v0.4.9_freebsd-amd64.tar.gz
+│   │   ├── go-ipfs_v0.4.9_linux-386.tar.gz
+│   │   ├── go-ipfs_v0.4.9_linux-amd64.tar.gz
+│   │   ├── go-ipfs_v0.4.9_linux-arm.tar.gz
+│   │   ├── go-ipfs_v0.4.9_windows-386.zip
+│   │   ├── go-ipfs_v0.4.9_windows-amd64.zip
+│   │   └── results
+│   └── versions
+└── index.html
+85 directories, 943 files
 ```
 
 We call this the **distribution index**, the listing of all distributions, their versions, and platform assets.
 
-Note how they each describe `<platform>` differently. This is likely to be inevitable as different platform identifiers will be used by different communities.
+## Install
 
-### distribution index versions / updates
+First, install the following dependencies via your favorite package manager:
 
-The **distribution index** changes over time, kind of like a git repository. [Since we don't yet have commits](https://github.com/ipfs/notes/issues/23), we will just do a poor-man's versioning for the index itself. We will write all version hashes to a file `versions` in this repository.
+* yarn
+* hugo
+* npm
+* jq
+* ipfs
+* git (obviously)
 
-A site like `dist.ipfs.io` or `ipfs.io/dist` would just serve the _latest_ version of the index.
-
-## Install
+Then install the javsacript dependencies with yarn:
 
 ```sh
-$ git clone https://github.com/ipfs/distributions
-$ cd distributions
 # Install javascript dependencies
-$ yarn
-# Install dependency for make files
-$ npm install -g jq
+> yarn
 ```
 
-This project uses a makefile + scripts to build all the things.
+Finally, run `make` to build/download the existing distribution over IPFS. It will download everything published at dist.ipfs.io and then build anything missing.
 
 ```sh
-$ make
+> make
 ```
 
-should do everything.
+## Usage
 
+### Adding a version
 
-## Usage
+Run:
+
+```sh
+> ./dist.sh add-version <dist> <version>
+```
+
+This will add the version to `dists/<dist>/versions`, set it as the current version in `dists/<dist>/current`, and build it.
+
+### Adding a new (go) distribution
+
+Run:
 
-Each `<dist>` has a directory in the root of this repo. inside it there is a `Makefile` and other necessary scripts. Running
+```sh
+> ./dist.sh new-go-dist <dist> <git-repo>
+```
+
+And follow the prompts.
+
+### Publishing 
 
+In the root of the repository, run:
+
+```sh
+> make publish
 ```
-make
+
+This will build (or download) anything that hasn't been built and build, compile the index in `releases`, and add releases to ipfs. Save the hash it spits out, that's the new hash for `dists.ipfs.io`.
+
+Next, you should probably compare this with the current `dists.ipfs.io` to make sure nothing is amiss:
+
+```sh
+> ipfs object diff /ipns/dist.ipfs.io /ipfs/<NEW_HASH>
 ```
 
-Should:
```

Comment:
I would also recommend you actually open the hash in your browser `http://127.0.0.1:8080/ipfs/HASH` and make sure that the latest version shows up and in general things look okay.